medicines/pars-upload/src/wizard.js

```diff
@@ -1,5 +1,62 @@
-import { useState } from "react";
+import { useState } from 'react'
 
-export const Wizard = () => {
-  const [pageIndex, setPageIndex] = useState(0);
-};
+export const Wizard = ({ initialSteps, success: Success, onComplete }) => {
+  const [steps, setSteps] = useState(() =>
+    initialSteps.map(({ type, component }) => ({ type, component, data: null }))
+  )
+  const [pageIndex, setPageIndex] = useState(0)
+
+  const currentPage = steps[pageIndex]
+  const Component = currentPage.component
+
+  const setPageData = (pageData) =>
+    setSteps((steps) =>
+      steps.map((page, i) =>
+        i === pageIndex ? { ...page, data: pageData } : page
+      )
+    )
```

Comment:
I think this looks good but it's a complex state machine so might call you with a few questions :)